oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtAudienceValidator.java

```diff
@@ -32,19 +32,30 @@ public final class JwtAudienceValidator implements OAuth2TokenValidator<Jwt> {
 
 	private final JwtClaimValidator<Collection<String>> validator;
 
+	private boolean allowEmpty;
+
 	/**
 	 * Constructs a {@link JwtAudienceValidator} using the provided parameters
 	 * @param audience - The audience that each {@link Jwt} should have.
 	 */
 	public JwtAudienceValidator(String audience) {
 		Assert.notNull(audience, "audience cannot be null");
+		this.allowEmpty = false;
 		this.validator = new JwtClaimValidator<>(JwtClaimNames.AUD,
-				(claimValue) -> (claimValue != null) && claimValue.contains(audience));
+				(claimValue) -> (claimValue != null) ? claimValue.contains(audience) : allowEmpty);
+	}
+
+	/**
+	 * Whether to allow the {@code aud} claim to be empty. The default value is
+	 * {@code false}
+	 */
+	public void setAllowEmpty(boolean allowEmpty) {
+		this.allowEmpty = allowEmpty;
 	}
 
 	@Override
 	public OAuth2TokenValidatorResult validate(Jwt token) {
-		Assert.notNull(token, "token cannot be null");
```

Comment:
I think the intent is to say the parameter name, not the datatype. Will you please leave this as-is?